Ginger/Ginger/ALM/ALMConnectionPage.xaml.cs

```diff
@@ -612,32 +671,37 @@ private void ServerURLTextBox_TextChanged(object sender, TextChangedEventArgs e)
 
         private void UserNameTextBox_TextChanged(object sender, TextChangedEventArgs e)
         {
-             WorkSpace.Instance.UserProfile.ALMUserName = UserNameTextBox.Text;
+            //WorkSpace.Instance.UserProfile.ALMUserName = UserNameTextBox.Text;
             SetControls();
         }
 
         private void PasswordTextBox_PasswordChanged(object sender, RoutedEventArgs e)
         {
-            ALMIntegration.Instance.SetALMPassword(PasswordTextBox.Password);
+            CurrentAlmUserConfigurations.ALMPassword = PasswordTextBox.Password;
+            //ALMIntegration.Instance.SetALMPassword(PasswordTextBox.Password);
             SetControls();
         }
 
         private void ProjectComboBox_SelectionChanged(object sender, SelectionChangedEventArgs e)
         {
             if (ProjectComboBox != null && ProjectComboBox.SelectedItem != null)
             {
-                ALMIntegration.Instance.SetALMProject((KeyValuePair<string, string>)ProjectComboBox.SelectedItem);
+                ALMIntegration.Instance.SetALMProject((KeyValuePair<string, string>)ProjectComboBox.SelectedItem,CurrentAlmConfigurations.AlmType);
                 return;
             }
             if (ProjectComboBox != null && ProjectComboBox.SelectionBoxItem != null)
             {
                 ProjectComboBox.SelectedItem = ProjectComboBox.SelectionBoxItem;
-                ALMIntegration.Instance.SetALMProject((KeyValuePair<string, string>)ProjectComboBox.SelectionBoxItem);
+                ALMIntegration.Instance.SetALMProject((KeyValuePair<string, string>)ProjectComboBox.SelectionBoxItem, CurrentAlmConfigurations.AlmType);
             }
         }
 
         private void TestALMConnectionButton_Click(object sender, RoutedEventArgs e)
         {
+            if (CurrentAlmConfigurations != null)
```

Comment:
What if currert ALM configurations is  null  ? it will execute code after if and will fail or throw exception no ?